syntax = "proto3";

package shipping.interfaces;

service OrderService {
    rpc createOrder(CreateOrderRequest) returns (CreateOrderResponse);
    rpc getOrder(GetOrderRequest) returns (GetOrderResponse);
    rpc updateOrder(UpdateOrderRequest) returns (UpdateOrderResponse);
}

message Order {
  string statusPurchase = 1;
  string buyer = 2;
  string name = 3;
  float price = 4;
  string address = 5;
  string phoneBuyer = 6;
  string cityDelivery = 7;
}

message CreateOrderRequest {
    Order order = 1;
}

message CreateOrderResponse {
    string messageOfConfirmation = 1;
}

message GetOrderRequest {
    string name = 1;
}

message GetOrderResponse {
    Order order = 1;
}

message UpdateOrderRequest {
    string name = 1;
    Order order = 2;
}

message UpdateOrderResponse {
    string messageOfConfirmation = 1;
}
